netmap/types.proto

```diff
@@ -130,6 +130,53 @@ message PlacementPolicy {
     json_name = "subnetId",
     deprecated = true
   ];
+
+  // Erasure coding rule for container objects.
+  //
+  // For each original object, the payload is split into `data_part_num` data
+  // and `parity_part_num` parity parts. Each part is the same size. Data parts
+  // contain the original payload. If its length is not divisible by
+  // `data_part_num`, the last part is aligned with zero bytes. Both
+  // `data_part_num` and `parity_part_num` MUST NOT be zero or exceed 64,
+  // including in total.
+  //
+  // For each payload part, a part object is created. Original object's ID,
+  // signature and header is written in `header.split.parent`,
+  // `header.split.parent_signature` and `header.split.parent_header` fields
+  // correspondingly. Part index is written in the `__NEOFS__EC_PART_IDX`
+  // attribute as base-10 integer. Rule index in `PlacementPolicy.ec_rules`
+  // list is written in the `__NEOFS__EC_RULE_IDX` attribute as base-10
+  // integer.
+  //
+  // Each part object is stored in the container in one copy. Storage nodes are
+  // selected from the network map similar to `PlacementPolicy.replicas` rules.
+  // Optional `selector` acts the same way. The object for the `i`-th part is
+  // placed in the `i`-th node. If it is unavailable, the backup nodes with
+  // indexes `m * n + i` (`n = data_part_num + parity_part_num`,
+  // `m = 1, ..., CBF-1`). If all nodes for the `i`-th part
+  // are unavailable, nodes for the `i+1`-th (0 for the last) part are tried,
+  // and so on.
+  //
+  // Once part objects are stored in the container, the original object remains
+  // available if at least `data_part_num` of any part objects are available.
+  // In other words, unavailability (including complete loss) of any of
+  // `parity_part_num` part objects does not violate availability of the
+  // original one.
+  //
+  // Objects of TOMBSTONE and LOCK types are not encoded and stored as they are
+  // because they have no payload.
+  message ECRule {
+    // Number of data parts
+    uint32 data_part_num = 1;
+
+    // Number of parity parts
+    uint32 parity_part_num = 2;
+
+    // Name of the linked selector
+    string selector = 3 [json_name = "selector"];
+  }
+  // Erasure coding rules. Limited to 256 items.
+  repeated ECRule ec_rules = 6 [json_name = "ecRules"];
```

Comment:
Also it's completely separated from `replicas` which makes these rules kinda alternative. It can be an implementation detail, but the fact that they're not on the same level can be confusing.

Reply:
they are at the same level of policy, just diff fields. I though that we can add `parity` field to `Replica`. Zero stands for old REP rule, non-zero - for EC one. But the storage scheme is so different, so i decided not to merge them. And IINM there wont be any gain in size

Reply:
As discussed yesterday policy string representation is already a mix of different things from the same level (rep/cbf/selectors) and it has no order logically. Since EC is a new features that differs significantly in behavior from replicas having another field in protobuf is more convenient in many cases.